### PR TITLE
[GEOS-8052] Support virtual services for CSW GetCapabilities

### PR DESCRIPTION
**Pull request by tbarsballe**
Fix for https://osgeo-org.atlassian.net/browse/GEOS-8052

GeoServer.getService already has logic to deal with virtual services, so that just needed to be hooked up.

### CODE REVIEW

File: src/extension/csw/core/src/main/java/org/geoserver/csw/DefaultWebCatalogService.java [hunk 1]
```diff
@@ -24,6 +24,7 @@
 import org.geoserver.csw.response.CSWRecordsResult;
 import org.geoserver.csw.store.CatalogStore;
 import org.geoserver.csw.store.RepositoryItem;
+import org.geoserver.ows.LocalWorkspace;
 import org.geoserver.platform.GeoServerExtensions;
```
Comment (nmco) [inline src/extension/csw/core/src/main/java/org/geoserver/csw/DefaultWebCatalogService.java]: Is this import really needed ?

File: src/extension/csw/core/src/test/java/org/geoserver/csw/GetCapabilitiesTest.java [hunk 2]
```diff
@@ -168,6 +172,39 @@ public void testPostBasic() throws Exception {
                         dom));
     }
 
+    @Test
+    public void testVirtualService() throws Exception {
+        List<CSWInfo> infos = GeoServerExtensions.extensions(CSWInfo.class);
+
+
```
Comment (nmco) [inline src/extension/csw/core/src/test/java/org/geoserver/csw/GetCapabilitiesTest.java]: Extra blank line.

File: src/extension/csw/core/src/test/java/org/geoserver/csw/GetCapabilitiesTest.java [hunk 3]
```diff
@@ -168,6 +172,39 @@ public void testPostBasic() throws Exception {
...
+        // Test global abstract
+        Document dom = getAsDOM(BASEPATH + "?service=csw&version=2.0.2&request=GetCapabilities");
+        print(dom);
+        checkValidationErrors(dom);
+        assertEquals("Global CSW", xpath.evaluate("//ows:ServiceIdentification/ows:Abstract", dom));
+
+        //Test local abstract
+        dom = getAsDOM("gs/" + BASEPATH + "?service=csw&version=2.0.2&request=GetCapabilities");
+        print(dom);
```
Comment (nmco) [inline src/extension/csw/core/src/test/java/org/geoserver/csw/GetCapabilitiesTest.java]: You may want to remove the print.

**Comment by nmco** [comment]
Minor comments, looks good.

File: src/extension/csw/core/src/main/java/org/geoserver/csw/DefaultWebCatalogService.java (hunk 1 above)
Comment (tbarsballe) [inline src/extension/csw/core/src/main/java/org/geoserver/csw/DefaultWebCatalogService.java]: Nope, forgot to clean up imports after finalizing a solution. Will fix.

File: src/extension/csw/core/src/test/java/org/geoserver/csw/GetCapabilitiesTest.java (hunk 3 above)
Comment (tbarsballe) [inline src/extension/csw/core/src/test/java/org/geoserver/csw/GetCapabilitiesTest.java]: Yes. I should probably also remove it from the test I copied it from.

**Comment by tbarsballe** [comment]
Updated based on feedback

**Comment by nmco** [comment]
Thanks, I guess this should be backport to 2.11.x and 2.10.x.

**Comment by tbarsballe** [comment]
Backported to 2.11.x and 2.10.x

**Pull request merged**